addons/purchase_stock/models/stock_rule.py

```diff
@@ -40,6 +40,7 @@ def _run_buy(self, procurements):
             schedule_date = (procurement_date_planned - relativedelta(days=procurement.company_id.po_lead))
 
             supplier = procurement.product_id._select_seller(
+                partner_id=procurement.values.get("partner"),
```

Comment:
i would name it "partner_id" even if it's a recordset, it is consistent with what we did in this file

Reply:
I cannot be used, as it is already used here: https://github.com/odoo/odoo/blob/13.0/addons/purchase_stock/models/stock_rule.py#L281

I renamed to `force_partner_id`, does it looks good to you?

Reply:
maybe supplier_id ?

Reply:
Ok, let's go for supplier_id :)